### PR TITLE
EODHP-997 - add api

### PR DESCRIPTION
**Pull request by hcollingwood**

### CODE REVIEW

File: annotations_api/api.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+app = FastAPI(root_path=root_path)
+
+annotations_bucket = os.environ.get("ANNOTATIONS_BUCKET", "")
+
+
+def key_to_json(key: str, request: Request) -> json:
+
+    base_url = str(request.base_url)
+    base_url = base_url if not base_url.endswith("/") else base_url[:-1]
```
Comment (apalmer-tpzuk) [inline annotations_api/api.py]: Could instead use `base_url = base_url.rstrip("/")`

File: annotations_api/api.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    url = f"{base_url}/{key}"
+
+    return {"file": key.split("/")[-1], "href": url}
+
+
+def bucket_contents_to_json(contents: dict, path: str, request: Request) -> json:
```
Comment (apalmer-tpzuk) [inline annotations_api/api.py:29]: Probably still worth putting docstrings in, even though function names are descriptive enough for what they do

File: annotations_api/api.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    file_name = f"{uuid}.{file_type}" if file_type else uuid
+
+    key = f"catalogue/{path}/annotations/{file_name}"
+
+    try:
+        data = s3.get_object(Bucket=annotations_bucket, Key=key)
+
+    except ClientError:
+        try:
+            data = s3.get_object(Bucket=annotations_bucket, Key=f"{key}.ttl")
```
Comment (apalmer-tpzuk) [inline annotations_api/api.py:71]: Not sure about this logic in a couple of cases - specifically when there is no file type defined, and if the original requested file type does not exist. Might be best to discuss exactly what functionality we want here.

File: annotations_api/api.py (hunk 1 above)
Comment (hcollingwood) [reply annotations_api/api.py]: Good idea

File: annotations_api/api.py (hunk 2 above)
Comment (hcollingwood) [reply annotations_api/api.py:29]: Updated

File: annotations_api/api.py (hunk 3 above)
Comment (apalmer-tpzuk) [reply annotations_api/api.py:71]: As discussed, logic is intended

**Review by apalmer-tpzuk — APPROVED** [review]

**Pull request merged**